ingestion/src/metadata/ingestion/source/database/datalake.py

```diff
@@ -0,0 +1,325 @@
+from metadata.generated.schema.entity.data.databaseSchema import DatabaseSchema
+from metadata.generated.schema.entity.data.table import Column, Table, TableData
+from metadata.generated.schema.entity.services.connections.database.datalakeConnection import (
+    DatalakeConnection,
+)
+from metadata.generated.schema.entity.services.connections.metadata.openMetadataConnection import (
+    OpenMetadataConnection,
+)
+from metadata.generated.schema.entity.services.databaseService import DatabaseService
+from metadata.generated.schema.metadataIngestion.databaseServiceMetadataPipeline import (
+    DatabaseServiceMetadataPipeline,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as WorkflowSource,
+)
+from metadata.generated.schema.type.entityReference import EntityReference
+from metadata.ingestion.api.common import Entity
+from metadata.ingestion.api.source import InvalidSourceException, Source, SourceStatus
+from metadata.ingestion.models.ometa_table_db import OMetaDatabaseAndTable
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.ingestion.source.database.common_db_source import SQLSourceStatus
+from metadata.utils.column_type_parser import ColumnTypeParser
+from metadata.utils.connections import get_connection
+from metadata.utils.filters import filter_by_table
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+class DatalakeSource(Source[Entity]):
+    def __init__(self, config: WorkflowSource, metadata_config: OpenMetadataConnection):
+        super().__init__()
+        self.status = SQLSourceStatus()
+
+        self.config = config
+        self.source_config: DatabaseServiceMetadataPipeline = (
+            self.config.sourceConfig.config
+        )
+        self.metadata_config = metadata_config
+        self.metadata = OpenMetadata(metadata_config)
+        self.service_connection = self.config.serviceConnection.__root__.config
+        self.service = self.metadata.get_service_or_create(
+            entity=DatabaseService, config=config
+        )
+        self.connection = get_connection(self.service_connection)
+        self.client = self.connection.client
+
+    @classmethod
+    def create(cls, config_dict, metadata_config: OpenMetadataConnection):
+        config: WorkflowSource = WorkflowSource.parse_obj(config_dict)
+        connection: DatalakeConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, DatalakeConnection):
+            raise InvalidSourceException(
+                f"Expected DatalakeConnection, but got {connection}"
+            )
+        return cls(config, metadata_config)
+
+    def prepare(self):
+        pass
+
+    def next_record(self) -> Iterable[Entity]:
+        try:
+
+            bucket_name = self.service_connection.bucketName
+            if hasattr(self.service_connection.configSource, "gcsConfig"):
+                yield from self.get_gcs_files(bucket_name)
```

Comment:
hi, I think that this means that if we do not pass a `bucketName`, then the ingestion does not happen. However, the field is not marked as required and the default value `""` is not really helpful.

IMO here we could consider a couple of scenarios:
- Users want to fetch the data from a single bucket
- Users want to fetch data from all buckets
Then, if the `bucketName` is informed, we will just list from there. If it is not, we will first list all buckets, and then list objects from each bucket.

Also, we should allow users to specify any `prefix` key:
```
conn.list_objects(Bucket='bucket_name', Prefix='prefix_string')
```
This `prefixString` should also be a parameter in the JSON Schema which is not required.